### ISSUE TITLE
Random prop/tree collections

### ISSUE THREAD

**Issue by originalfoo** [algernon-A/BOB#32]
# Summary

Ability to create a collection of props or trees, and then be able to replace a prop or tree with that collection - one of the items would be chosen at random each replacement.

It's similar to replacement packs, but just defining a collection of replacements (rather than a specific mapping `A -> B`)

# Example use case:

I want to customise the foilage along macwelshman's [Rural Roads](https://steamcommunity.com/sharedfiles/filedetails/?id=1683326237), of which there are several. By default these roads use [deciduous](https://steamcommunity.com/workshop/filedetails/?id=1225368249) [trees](https://steamcommunity.com/sharedfiles/filedetails/?id=1501682221), but those look out of place when the road is going through a huge pine tree (evergreen) forest. 

So I want to replace the road's trees with various "scrub" trees from [Pine Scrub Pack](https://steamcommunity.com/sharedfiles/filedetails/?id=2064797149).

I could create a collection "Pine Scrub", add the trees from the scrub pack, set their probabilities, and then just use that collection as if it were a tree when defining replacemetns.

Note that it would be ideally be random per tree/prop, not per segment/building, otherwise repetition would be very obvious even within a single segment (based on my current testing). For example, see Klyte's "Prop Replacer" mod which allows "Fully Random" replacements as alternative to per-segment/building seed.

Another example is I could create a collection of map-themed trees to use in buildings, then not have to worry about choosing specific tree when replacing vanilla trees in my buildings. Same sort of thing with props, I could just bundle, for example, a load of residential garden props in to a collection and have low density residential buildings use those.

# How it could work:

* Create collection button - user specifies collection name and type (tree or prop)
* Select which trees/props go in the collection
* Save collection
* Collections then appear in the replacements list alongside normal trees/props (as applicable)
    * Eg. if replacing a tree, the list in right panel would include collections ( maybe in different colour so they stand out )
    * Maybe even allow collections within collections (with recursion prevention)?
        * Eg. I could create small collections of trees, then bundle those in to a 'forest' collection
    * Could possibly make tree collections available for use in forest brush and vice versa?
* Would need some way to edit collections (eg. if I want to add/remove trees to/from my pine scrub collection)
    * When collections are edited, refresh replacements using it city-wide
* Note: Doesn't need to be called a "Collection", could be a "Asset Pack" or "Group" or whatever

# Some notable benefits:

* I can put assets from different authors / assets in to a collection
* Saves time because I can now just say "replace with something from this collection"
* Use in multiple assets (eg. there are about 12 rural roads where I want pine scrub)
* I can edit the collection contents to propagate changes throughout my city wherever that collection is used
* When replacing map trees, I could replace them with a collection - eg. if a map was filled with pine trees, I could create a collection with a mixture of deciduous trees and now I've got a much more interesting forest... and if I edit the collection, could potentially refresh my forest? (would need some way to remember original map trees which is likely painful tho?)

**Comment by originalfoo** [comment]
Oh, wait, the "Random Setup" button already does this? (I thought it just chose random replacements, rather than create a 'collection' style thing). Doh!

**Comment by algernon-A** [comment]
I just saw this and was about to mention the 'random replacements'!  UI can do with a bit more development, but yes, the functionality is there.

That being said, I'm going to keep this open for now, as I think what you've written above will go great in the documentation I'm starting to pull together - so thanks!

**Comment by originalfoo** [comment]
I just realised why I was struggling with the existing random prop/tree functionality.... If I already had the 'show random' checkbox selected, then I go in and create my first random group, and exit the random window, the list in the main window doesn't update so it looks like my random group doesn't exist.

Suggestion: Refresh the right-hand list in main UI after random groups are added/removed/renamed. Maybe also show random groups in the main list but prefixed with `[r]` and in different colour?

(BTW, now I've got it working, the random feature is really awesome! Already created several random tree groups for roads, buildings, etc., love it!)
